### PR TITLE
Support DesignatedVoting when filtering events

### PR DESCRIPTION
**Pull request by ptare**
This PR also removes the method `Voting::hasRevealed`, is there a reason
to keep it around? If so, we'd have to either 1) add it to
`DesignatedVoting.sol`, or 2) have it take in a voter address rather
than relying on `msg.sender`.

Issue #787 

Signed-off-by: Prasad Tare <prasad.tare@gmail.com>

### CODE REVIEW

File: core/contracts/Voting.sol [hunk 1]
```diff
@@ -348,23 +348,6 @@ contract Voting is Testable, Ownable, OracleInterface, VotingInterface, Encrypte
         return voteTiming.computeCurrentRoundId(getCurrentTime());
     }
 
-    /**
-     * @notice Whether the caller has a revealed vote for the current round for an (identifier, time).
-     * @dev If the price request was resolved in a previous round, this function will return `false` even if the caller
-     * did reveal a vote.
-     */
-    function hasRevealedVote(bytes32 identifier, uint time) external view returns (bool) {
```
[the remark this answers is not in the document]
Comment (ptare) [reply core/contracts/Voting.sol]: Seems right. Do you think we should leave it in?

[the remark this answers is not in the document]
Comment (mrice32) [reply core/contracts/Voting.sol]: It *seems* fine to remove. We should be able to get all of this information from events, right?

[the remark this answers is not in the document]
Comment (ptare) [reply core/contracts/Voting.sol]: Yeah I think so.

File: voter-dapp/src/Dashboard.js [hunk 2]
```diff
@@ -44,25 +44,26 @@ function Dashboard() {
...
       votingGateway = deployedDesignatedVotingAddress;
-      votingAccount = deployedDesignatedVotingAddress;
     } else {
       // Waiting for drizzle finish loading DesignatedVoting.
       return <div>LOADING</div>;
     }
   }
+  // If commits/reveals are through a DesignatedVoting instance, then the address of the DesignatedVoting is the
+  // voting account as far as `Voting.sol` is concerned.
+  const votingAccount = votingGateway === "Voting" ? account : votingGateway;
```
Comment (mrice32) [inline voter-dapp/src/Dashboard.js]: +1. I like that this cannot be "" anymore.

**Pull request merged**